diff -uNr qemu-4.2.0/ui/keymaps.c qemu-4.2.0.mod/ui/keymaps.c
--- qemu-4.2.0/ui/keymaps.c	2019-12-12 20:20:48.000000000 +0200
+++ qemu-4.2.0.mod/ui/keymaps.c	2019-12-15 02:42:13.858750260 +0200
@@ -206,7 +206,7 @@
     keysym2code = g_hash_table_lookup(k->hash, GINT_TO_POINTER(keysym));
     if (!keysym2code) {
         trace_keymap_unmapped(keysym);
-        warn_report("no scancode found for keysym %d", keysym);
+        //warn_report("no scancode found for keysym %d", keysym);
         return 0;
     }
 
